### ISSUE TITLE
prefer-const no working

### ISSUE THREAD

**Issue by Soya-xy** [vuejs/eslint-plugin-vue#1705]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [ ] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:*^8.2.0" 
- **eslint-plugin-vue version:*^7.14.0* 
- **Node version:*v16.4.2* 
- **Operating System:*os X* 

**Please show your full configuration:**

```yml
env:
  browser: true
  es2021: true
  node: true
extends:
  - eslint:recommended
  - plugin:vue/vue3-recommended
  - prettier
parserOptions:
  ecmaVersion: 2022
  sourceType: module
plugins:
  - vue
rules:
  vue/require-default-prop: 0
  vue/attribute-hyphenation: 0
```

**What did you do?**	

```js
let a = [1,2,3]
a.map(v=>v++)
```
prefer-const no working
Here esLint should report an error and use const instead of let

**What did you expect to happen?**

**What actually happened?**

**Repository to reproduce this issue**
